src/localconf.js

```diff
@@ -82,11 +84,37 @@ function buildConfig(opts) {
   const submissionApiPath = `${hostname}api/v0/submission/`;
   const apiPath = `${hostname}api/`;
   const graphqlPath = `${hostname}api/v0/submission/graphql/`;
+  const peregrineVersionPath = `${hostname}api/search/_version`;
   const dataDictionaryTemplatePath = `${hostname}api/v0/submission/template/`;
   let userAPIPath = typeof fenceURL === 'undefined' ? `${hostname}user/` : ensureTrailingSlash(fenceURL);
   const jobAPIPath = `${hostname}job/`;
   const credentialCdisPath = `${userAPIPath}credentials/cdis/`;
-  const coreMetadataPath = `${hostname}coremetadata/`;
+
+  const coreMetadataPathPromise = fetch(peregrineVersionPath)
+    .then((response) => response.text())
+    .then((responseBody) => {
+      // if peregrine is on version 3.2.0/2023.04 or newer, or on a branch, use
+      // the peregrine endpoint. if not, use the deprecated pidgin endpoint
+      const minSemVer = '3.2.0';
+      const minMonthlyRelease = semver.coerce('2023.04.0', { loose: true });
+      const monthlyReleaseCutoff = semver.coerce('2020', { loose: true });
+
+      var peregrineVersion = JSON.parse(responseBody).version;
+      var url = `${hostname}api/search/coremetadata/`;
+      if (peregrineVersion) {
+        try {
+          peregrineVersion = semver.coerce(peregrineVersion, { loose: true });
+          if (
+            semver.lt(peregrineVersion, minSemVer) ||
+            (semver.gte(peregrineVersion, monthlyReleaseCutoff) && semver.lt(peregrineVersion, minMonthlyRelease))
+          ) {
+            url = `${hostname}coremetadata/`; // pidgin endpoint
+          }
+        } catch (error) { } // can't parse or compare the peregrine version: don't use legacy url
+      }
+      return url;
+    });
```

Comment:
jset seems to be not happy with the format here

Reply:
oh wait... it might be that jest dones't like we return a promise from this function 🤦 

if that is the case, maybe we should move it back, sorry

Reply:
yeah i've been struggling with this since yesterday, jest really doesn't like this 🤔 maybe there is somewhere else i could move it so the call is only made once